File: source code/front-end-user/src/app/models/user-update.ts
import { UserProfile } from './user-profile';

export class UserUpdate {
  username: string;
  password?: string;
  profile: UserProfile;
  birthDate?: Date; // ✅ Thêm ngày sinh, kiểu ISO string

  constructor(
    username: string,
    profile: UserProfile,
    birthDate?: Date,
    password?: string
  ) {
    this.username = username;
    this.profile = profile;
    this.birthDate = birthDate;
    this.password = password;
  }
}

File: source code/backend/src/main/java/com/thanhtam/backend/dto/UserUpdate.java
package com.thanhtam.backend.dto;

import java.util.Date;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.thanhtam.backend.entity.Profile;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserUpdate {

    private String username;
    private String password;
    private Profile profile;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd", timezone = "UTC")
    private Date birthDate;
}
